### PR TITLE
[CELEBORN-819] Worker close should pass close status to support handle graceful shutdown and decommission

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Pass exit kind to each component, if the exit kind match:

- GRACEFUL_SHUTDOWN: Behavior as origin code's graceful == true
- Others: will clean the level db file.

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @waitinfuture @pan3793

File: common/src/main/scala/org/apache/celeborn/common/util/CelebornExitStatus.scala [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.celeborn.common.util
+
+private[celeborn] object CelebornExitStatus {
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/util/CelebornExitStatus.scala:20]: `Status` is not suitable, maybe `CelebornExitKind`

File: common/src/main/scala/org/apache/celeborn/common/util/CelebornExitStatus.scala [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  val EXIT = 0
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/util/CelebornExitStatus.scala:21]:
```suggestion
  val EXIT_IMMEDIATELY = 0
```

File: common/src/main/java/org/apache/celeborn/common/network/server/TransportServer.java [hunk 3]
```diff
@@ -130,24 +131,24 @@ protected void initChannel(SocketChannel ch) {
 
   @Override
   public void close() {
-    shutdown(true);
+    shutdown(CelebornExitStatus.WORKER_GRACEFUL_SHUTDOWN());
   }
 
-  public void shutdown(boolean graceful) {
+  public void shutdown(int exitCode) {
```
Comment (pan3793) [inline common/src/main/java/org/apache/celeborn/common/network/server/TransportServer.java:137]: `exitCode` is definitely misleading, it usually represents the process exit code. Use `exitKind` or simply `kind` instead.

Comment (AngersZhuuuu) [reply common/src/main/java/org/apache/celeborn/common/network/server/TransportServer.java]: Done

File: common/src/main/scala/org/apache/celeborn/common/util/CelebornExitStatus.scala (hunk 2 above)
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/util/CelebornExitStatus.scala]: Done

File: common/src/main/scala/org/apache/celeborn/common/util/CelebornExitStatus.scala (hunk 1 above)
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/util/CelebornExitStatus.scala]: Done

File: service/src/main/scala/org/apache/celeborn/server/common/http/HttpServer.scala [hunk 4]
```diff
@@ -56,7 +56,7 @@ class HttpServer(
     isStarted = true
   }
 
-  def stop(graceful: Boolean): Unit = synchronized {
+  def stop(exitCode: Int): Unit = synchronized {
```
Comment (pan3793) [inline service/src/main/scala/org/apache/celeborn/server/common/http/HttpServer.scala:59]:
```suggestion
  def stop(exitKind: Int): Unit = synchronized {
```

**Review by pan3793 — APPROVED** [review]

**Review by cfmcgrady — APPROVED** [review]

File: common/src/main/scala/org/apache/celeborn/common/util/CelebornExitKind.scala [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.celeborn.common.util
+
+private[celeborn] object CelebornExitKind {
+  val EXIT_IMMEDIATELY = 0
+  val WORKER_GRACEFUL_SHUTDOWN = 1
+  val WORKER_DECOMMISSION = 2
```
Comment (FMX) [inline common/src/main/scala/org/apache/celeborn/common/util/CelebornExitKind.scala:23]: What's the difference between EXIT_IMMEDIATELY and WORKER_DECOMMISSION?  Will WORKER_DECOMMISSION waiting for current shuffle end?

Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/util/CelebornExitKind.scala:23]: Yea.

**Pull request closed without merging**